File: cab/cab/Models/Cab.swift
import Foundation

struct Cab: Codable, Identifiable {
    let id: String
    let driverName: String
    let driverPhone: String
    let vehicleModel: String
    let licensePlate: String
    let color: String
    let seaterCapacity: Int
    let isCNG: Bool
    let isActive: Bool
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case driverName, driverPhone, vehicleModel, licensePlate, color
        case seaterCapacity, isCNG, isActive, createdAt
    }
}

struct CreateCabRequest: Encodable {
    let driverName: String
    let driverPhone: String
    let vehicleModel: String
    let licensePlate: String
    let color: String
    let seaterCapacity: Int
    let isCNG: Bool
}

File: backend/src/models/Cab.js
const mongoose = require('mongoose');

const cabSchema = new mongoose.Schema(
  {
    driverName: { type: String, required: true, trim: true },
    driverPhone: { type: String, required: true, trim: true },
    vehicleModel: { type: String, required: true, trim: true },
    licensePlate: { type: String, required: true, unique: true, uppercase: true, trim: true },
    color: { type: String, required: true, trim: true },
    seaterCapacity: { type: Number, required: true, enum: [4, 6, 7] },
    isCNG: { type: Boolean, default: false },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Cab', cabSchema);
